core/src/main/scala/kafka/log/Log.scala

```diff
@@ -230,13 +230,6 @@ class Log(@volatile var dir: File,
   }
 
   def updateConfig(updatedKeys: Set[String], newConfig: LogConfig): Unit = {
-    if ((updatedKeys.contains(LogConfig.RetentionMsProp)
-      || updatedKeys.contains(LogConfig.MessageTimestampDifferenceMaxMsProp))
-      && topicPartition.partition == 0  // generate warnings only for one partition of each topic
-      && newConfig.retentionMs < newConfig.messageTimestampDifferenceMaxMs)
```

Comment:
99% of the time, the max timestamp difference is Long.MaxValue. It's weird to log a warning for the default setting. We could add a special case, but the benefit of this message seems marginal at best.